src/components/crossDevice/MobileFlow.js

```diff
@@ -33,15 +36,22 @@ class MobileFlow extends Component {
 
   onClientSuccess = () => {
     this.props.actions.setClientSuccess(true)
+    this.props.actions.mobileConnected(false)
   }
 
   onDisconnectPing = (data) => {
     this.sendMessage('disconnect pong', data.roomId)
+    this.props.actions.mobileConnected(false)
   }
 
-  render = (props) =>
-    this.props.clientSuccess ?
-      <CrossDeviceSubmit {...props}/> : <MobileConnected {...props}/>
+  render = (props) => {
+    if (this.props.clientSuccess)
+      return <CrossDeviceSubmit {...props}/>
+    if (this.props.mobileConnected)
```

Comment:
There are some bugs that should be simple to fix, but overall I think keeping track of whether the mobile is connected is a much better way to decide what screen to show compared to using the mobile number!